File: app/src/main/java/io/github/tobyhs/weatherweight/data/LocationNotFoundError.java
package io.github.tobyhs.weatherweight.data;

/**
 * This exception is thrown when a {@code WeatherRepository} can't find a given location.
 */
public class LocationNotFoundError extends Exception {
    /**
     * @param location the location that was not found
     */
    public LocationNotFoundError(String location) {
        super("Location " + location + " not found");
    }
}

File: app/src/main/java/io/github/tobyhs/weatherweight/data/LocationNotFoundError.kt
package io.github.tobyhs.weatherweight.data

/**
 * This exception is thrown when a `WeatherRepository` can't find a given location.
 */
class LocationNotFoundError
/**
 * @param location the location that was not found
 */
    (location: String) : Exception("Location $location not found")
